fe/fe-core/src/main/java/com/starrocks/common/util/DurationStyle.java

```diff
@@ -0,0 +1,109 @@
+
+package com.starrocks.common.util;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.Objects;
+
+import static org.joda.time.DateTimeConstants.SECONDS_PER_DAY;
+import static org.joda.time.DateTimeConstants.SECONDS_PER_HOUR;
+import static org.joda.time.DateTimeConstants.SECONDS_PER_MINUTE;
+
+public abstract class DurationStyle {
+    public static final DurationStyle LONG = new LongStyle();
+    public static final DurationStyle ISO8601 = new ISO8601Style();
+
+    public abstract Duration parse(String text);
+
+    public abstract String toString(Duration duration);
+
+    private static class ISO8601Style extends DurationStyle {
+
+        @Override
+        public Duration parse(String text) {
+            return Duration.parse(text);
+        }
+
+        @Override
+        public String toString(Duration duration) {
+            return duration.toString();
+        }
+    }
+
+    private static class LongStyle extends DurationStyle {
+
+        @Override
+        public Duration parse(String text) {
+            Objects.requireNonNull(text, "text is null");
+            String[] tokens = text.trim().split("\\s+");
+            if ((tokens.length == 0) || (tokens.length % 2 != 0)) {
+                throw new DateTimeParseException("Cannot parse text to Duration", text, 0);
+            }
+            long seconds = 0;
+            for (int i = 0; i < tokens.length; i += 2) {
+                try {
+                    long number = Long.parseLong(tokens[i]);
+                    String unit = tokens[i + 1].toLowerCase();
+                    switch (unit) {
+                        case "second":
+                        case "seconds":
+                            seconds += number;
+                            break;
+                        case "minute":
+                        case "minutes":
+                            seconds += number * SECONDS_PER_MINUTE;
+                            break;
+                        case "hour":
+                        case "hours":
+                            seconds += number * SECONDS_PER_HOUR;
+                            break;
+                        case "day":
+                        case "days":
+                            seconds += number * SECONDS_PER_DAY;
+                            break;
+                        default:
+                            throw new DateTimeParseException("Cannot parse text to Duration", text, 0);
+                    }
+                } catch (NumberFormatException ignored) {
+                    throw new DateTimeParseException("Cannot parse text to Duration", text, 0);
+                }
+            }
+            return Duration.ofSeconds(seconds);
+        }
+
+        @Override
+        public String toString(Duration duration) {
+            if (duration.isZero()) {
+                return "0 second";
+            }
+            long seconds = duration.getSeconds();
```

Comment:
seconds may be bigger than SECONDS_PER_DAY?

Reply:
Yes, and if so the number of hours will be greater than 24, which is fine.